decavision/dataset_preparation/generate_pseudolabels.py

```diff
@@ -0,0 +1,159 @@
+import os
+import pandas as pd
+import tensorflow as tf
+import numpy as np
+import shutil
+import matplotlib.pyplot as plt
+
+from tensorflow.keras.models import load_model
+from distutils.dir_util import copy_tree
+from tqdm import tqdm
+
+
+class PseudoLabelGenerator:
+    """
+    Class to make pseudo labels.
+    """
+
+    def __init__(self):
+        pass
+
+    def _load_img(self, path, target_size=(299, 299)):
+        """
+        Load an image from a given path and normalize it
+
+        Args:
+            path (list): Input image path
+            target_size (tuple): Size of image
+        Returns:
+            np.array: Numpy array of the data
+        """
+        # Read image
+        bits = tf.io.read_file(path)
+        image = tf.image.decode_jpeg(bits, channels=3)
+        # Resize
+        image = tf.image.resize(image, size=[*target_size])
+        image = tf.reshape(image, [*target_size, 3])
+        image = tf.cast(image, tf.uint8)
+        # Normalize [0, 1]
+        image = tf.image.convert_image_dtype(image, dtype=tf.float32)
+        image = image.numpy()
+        return image
+
+    def generate_pseudolabel_data(self, model_path,
+                                  train_data_path, unlabeled_path,
+                                  pseudo_data_path, save_confidences_csv=False,
+                                  plot_confidences=False, threshold=None):
+        """ Convert image and label to tfrecord example.
+            Arguments:
+                model_path (str): path to trained model
+                train_path (str): path to training data
+                unlabeled_path (str): path to unlabeled data
+                pseudo_path (str): path to store train data and pseudo data
+                save_confidences_csv (boolean): Whether to save CSV file with pseudo confidences, default is False.
+                plot_confidences (boolean): Whether to plot confidence graphs for raw confidences and per class confidences.
+                threshold (float): Discard images with prediction below this confidence, default is None.
+
+            Returns:
+                pseudo_data_path: A folder with both labeled and pseudo labeled images.
+        """
+        # Copy the training/labeled data to the destination folder where
+        # we will also store the pseudo labels.
+        copy_tree(train_data_path, pseudo_data_path)
+
+        # Load model
+        model = load_model(model_path, compile=False)
+        print("Loaded model.")
+
+        # Make dictionary for classes and their index
```

Comment:
With the format that I am aiming to use for the different methods I think this would fit better in the __init__.

Reply:
Done!